core/coreapi/coreapi.go

```diff
@@ -87,8 +137,99 @@ func (api *CoreAPI) PubSub() coreiface.PubSubAPI {
 	return (*PubSubAPI)(api)
 }
 
+// WithOptions returns api with global options applied
+func (api *CoreAPI) WithOptions(opts ...options.ApiOption) (coreiface.CoreAPI, error) {
+	settings := api.parentOpts // make sure to copy
+	_, err := options.ApiOptionsTo(&settings, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	if api.nd == nil {
+		return nil, errors.New("cannot apply options to api without node")
+	}
+
+	n := api.nd
+
+	subApi := &CoreAPI{
+		nctx: n.Context(),
+
+		identity:   n.Identity,
+		privateKey: n.PrivateKey,
+
+		repo:       n.Repo,
+		blockstore: n.Blockstore,
+		baseBlocks: n.BaseBlocks,
+		pinning:    n.Pinning,
+
+		blocks: n.Blocks,
+		dag:    n.DAG,
+
+		peerstore:       n.Peerstore,
+		peerHost:        n.PeerHost,
+		namesys:         n.Namesys,
+		recordValidator: n.RecordValidator,
+		exchange:        n.Exchange,
+		routing:         n.Routing,
+
+		pubSub: n.PubSub,
+
+		nd:         n,
+		parentOpts: settings,
+	}
+
+	subApi.isOnline = func(allowOffline bool) error {
```

Comment:
Code nit: given that this doesn't return a boolean "is online", this should probably be renamed to `checkOnline`.